ComponentKit/Components/CKNetworkImageComponent.mm

```diff
@@ -125,6 +125,12 @@ - (void)setSpecifier:(CKNetworkImageSpecifier *)specifier
   if (CKObjectIsEqual(specifier, _specifier)) {
     return;
   }
+  
+  BOOL isShowingCurrentImageURL = self.image != _specifier.defaultImage && self.image != nil;
+  if (isShowingCurrentImageURL && CKObjectIsEqual(_specifier.url, specifier.url)) {
+    _specifier = specifier;
+    return;
```

Comment:
CropRect is not taken into consideration in the isEqual method right now, so it is not breaking any existing functionality. I don't think we need a state machine, but instead we could check every property individually to decide what actions we should take in the setSpecifier: method:
- If cropRect changes, call setNeedsLayout.
- If the URL changes, cancel and start a new download.
- If the defaultImage changes, update the image if we were showing a defaultImage.